app/jobs/tracks/time_chunk_processor_job.rb

```
@@ -0,0 +1,136 @@
+  end
+
+  def create_track_from_points_array(points)
+    return nil if points.size < 2
+
+    begin
+      # Calculate distance using Geocoder with validation
+      distance = Point.calculate_distance_for_array_geocoder(points, :km)
+
+      # Additional validation for the distance result
+      if !distance.finite? || distance < 0
+        Rails.logger.error "Invalid distance calculated (#{distance}) for #{points.size} points in chunk #{chunk_data[:chunk_id]}"
```

RuboCop (1.76.1)

[convention] Line is too long. [130/120]

(Layout/LineLength)